test/integration/vorpal-engine_test.go

```diff
@@ -3,15 +3,21 @@
 package integration
 
 import (
+	"encoding/json"
+	"fmt"
 	"testing"
 
-	"github.com/checkmarx/ast-cli/internal/commands"
 	errorConstants "github.com/checkmarx/ast-cli/internal/constants/errors"
 	commonParams "github.com/checkmarx/ast-cli/internal/params"
+	"github.com/checkmarx/ast-cli/internal/wrappers/configuration"
+	"github.com/checkmarx/ast-cli/internal/wrappers/grpcs"
+	"github.com/spf13/viper"
+	assert2 "github.com/stretchr/testify/assert"
 	"gotest.tools/assert"
 )
 
```

Comment:
you have to check also the flows:

- scan vorpal with SourcesFlag and without the VorpalLatestVersion flag when the engine exe file doesn't exist
- scan vorpal with agent different from ast-cli with ai-protection license and without (maybe one of them you can check only with mock as a unit test but we need both tests)
- file doesn't exist
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

Reply:
There is a unit test that check the license case.
Will add the other tests  you mentioned.

Reply:
you have to add at least one integration test of the license
the integration test validates the struct you expected to get from the token.
if someone will change the token struct - this test will fail! and it will be great.